src/app/components/Table/Table.tsx

```diff
@@ -0,0 +1,98 @@
+import React, { FunctionComponent } from 'react';
+import {
+  ActionsColumn,
+  IAction,
+  IRow,
+  IRowData,
+  TableComposable,
+  Tbody,
+  Td,
+  Th,
+  Thead,
+  Tr,
+} from '@patternfly/react-table';
+import { css } from '@patternfly/react-styles';
+
+interface TableColumn {
+  /** Column identifier */
+  accessor: string;
+  /** Displayed label */
+  label: string;
+  /** Custom function to be used to render differently all values on this column */
+  formatter?: (value: IRowData, row?: IRow) => string | IRowData;
+}
+
+interface TableProps {
+  /** It enables the presence of an action menu on the rows, with the given resolver */
+  actionResolver?: (rowData: IRow) => IAction[];
+  /** Accessible name for the table */
+  ariaLabel?: string;
+  /** Element to be appended after the `tbody` node */
+  children?: React.ReactNode;
+  /** Columns to display */
+  columns: TableColumn[];
+  /** list of additive css classes */
+  cssClasses?: string | string[];
+  /** Collection of cells to render */
+  rows: IRow[];
+  /** Style variant for the table */
+  variant?: 'compact' | undefined;
+}
+
+export const Table: FunctionComponent<TableProps> = ({
+  actionResolver,
+  ariaLabel = 'Table',
+  children,
+  columns,
+  cssClasses,
+  rows,
+  variant = 'compact',
+}) => {
+  const transformColumns = (columns: TableColumn[]) => {
+    return columns.map((column) => column.label);
+  };
+
+  const transformRows = (rows: IRowData[], columns: TableColumn[]) => {
+    return rows.map((objectRow) => {
+      return {
+        cells: columns.map((column) => {
+          const accessor = column.accessor;
+          const formatter = column.formatter ?? ((value: IRowData) => value);
+          return formatter(objectRow[accessor], objectRow);
+        }),
+        originalData: objectRow,
+      };
+    });
+  };
+
+  return (
+    <TableComposable
+      className={css(cssClasses)}
+      variant={variant}
+      aria-label={ariaLabel}
+    >
+      <Thead>
+        <Tr>
+          {transformColumns(columns).map((column) => (
+            <Th key={column}>{column}</Th>
+          ))}
+        </Tr>
+      </Thead>
+      <Tbody>
+        {transformRows(rows, columns).map((row, rowIndex) => (
+          <Tr key={row?.originalData?.id ?? rowIndex}>
+            {row?.cells?.map((cell, cellIndex) => (
+              <Td key={cellIndex}>{cell}</Td>
+            ))}
+            {actionResolver && (
+              <Td className="pf-c-table__action">
```

Comment:
What PF version do we use? Can't we just set `isActionCell` ?

https://www.patternfly.org/v4/components/table#td

Reply:
I already tried. Not able to set it:
![image](https://user-images.githubusercontent.com/22591802/159871333-8c2663df-7b4b-45bd-83b3-fbe8009fd08a.png)
We are using:

```
    "@patternfly/patternfly": "4.164.2",
    "@patternfly/react-core": "4.181.1",
    "@patternfly/react-icons": "4.32.1",
    "@patternfly/react-styles": "4.31.1",
    "@patternfly/react-table": "4.50.1",
```
It seems that `isActionCell` will set only the `CSS` class, looking at the documentation.

Anyway, even if you set `isActionCell`, you need to manually wrap `<ActionsColumn` inside the `td`, while using `TableComposable`.
![image](https://user-images.githubusercontent.com/22591802/159871653-11dbb2df-707d-41ad-b6d8-3fcdf35062a9.png)

Reply:
Ok, to me it seems it was added by this commit into patternfly

https://github.com/patternfly/patternfly-react/commit/eaaedc28cf54cca1a47aaa5c4f5f76a11d32864d

So we are using PF where the feature is not available?

Reply:
Correct @jomarko. It seems we are using a version where this attribute is not defined.

Reply:
is it worth to comment in code or report a jira, so we can "fix/use the feature" in future, when we upgrade `patternfly/react-table` library?

Reply:
In my opinion no, because `isActionCell` adds only one CSS class (`pf-c-table__action`) to `td` element:
![image](https://user-images.githubusercontent.com/22591802/159876021-72792987-6f8c-4656-94bd-1867c07158e3.png)

Reply:
my 2 cents, I agree with @jomarko. If we know there's a `isActionCell` prop coming to do the job, we should track a ticket to swap to it.

I don't know if it's the right thing to do but I see there's another option, to add actions with the `actions` prop of `Td`.
It will add the action css class but it's also adding other inline styles, so aesthetically it's not the same thing right now. It may get the correct right alignment with more recent versions of PF though.

By the way the PF version we currently use is `2021.16`

Reply:
Traced a defect: https://issues.redhat.com/browse/MGDOBR-489

Reply:
:+1: